### PR TITLE
add edit code block

### PR DESCRIPTION
**Pull request by Wattenberger**
to showcase using the edit OpenAI endpoint, which currently has no rate-limiting

image

### CODE REVIEW

File: src/blocks/file-blocks/edit/index.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  );
+}
+
+const syntaxHighlighterLanguageMap = {
+  JavaScript: "javascript",
+  TypeScript: "typescript",
+} as Record<string, string>;
+
+const HunkComponent = ({ hunk, language }: { hunk: any; language: string }) => {
+  console.log(hunk);
```
Comment (mattrothenberg) [inline src/blocks/file-blocks/edit/index.tsx:129]: Drop log

**Review by mattrothenberg — APPROVED** [review]
Works well on my end! Very nice

File: src/blocks/file-blocks/edit/index.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  const [newContent, setNewContent] = useState<string>("");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const language = getLanguageFromFilename(context.path.split("/").pop() || "");
+
+  const hunks = useMemo(() => {
+    if (!newContent) return [];
+    const hunks = parseDiff(
+      diffAsText(content, newContent, {
+        context: 100,
```
Comment (jaked) [inline src/blocks/file-blocks/edit/index.tsx:24]: you're trying to get the whole file in the diff context here? is there a way to get `diffAsText` to produce a single hunk?

Comment (Wattenberger) [reply src/blocks/file-blocks/edit/index.tsx:24]: it's sadly a pretty restrictive API - this is the best I could do to get all of the unchanged lines in the "diff", so they're at least visible. But I could spend another hour or so poking around and looking at alternative libs!

File: src/blocks/file-blocks/edit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  return (
+    <pre className="px-5 py-3 text-left">
+      {hunk.changes.map((change: Hunk["change"], i: number) => (
+        <Change key={i} change={change} language={language} />
+      ))}
+    </pre>
+  );
+};
+
+const Change = ({ change, language }: { change: Hunk; language: string }) => {
```
Comment (jaked) [inline src/blocks/file-blocks/edit/index.tsx:138]: this is cool, I think the diff view in `blocks` doesn't do syntax highlighting

**Review by jaked — APPROVED** [review]
💯 

There is so much interesting stuff in various blocks. I wonder if it would be worth building out a standard library of blocks helper code.

File: src/blocks/file-blocks/edit/index.tsx (hunk 2 above)
Comment (jaked) [reply src/blocks/file-blocks/edit/index.tsx:24]: no please do not! I was just curious about the magic number.

File: src/blocks/file-blocks/edit/index.tsx (hunk 3 above)
Comment (Wattenberger) [reply src/blocks/file-blocks/edit/index.tsx:138]: you're right! it's the same library, which is honestly a pain to work with. I took it a step farther here, but ended up doing the styling & rendering myself and am probably missing some edge cases (seeing how large their codebase is).

File: src/blocks/file-blocks/edit/index.tsx (hunk 2 above)
Comment (Wattenberger) [reply src/blocks/file-blocks/edit/index.tsx:24]: it's a very valid question! definitely worth more exploration at some point :)

**Comment by Wattenberger** [comment]
great idea - might be worth adding to our `/utils` library at some point!

**Pull request merged**